### PR TITLE
node: collapse initialization internals

### PR DESCRIPTION
**Pull request by tychoish**
This has some (I think reasonable) reordering effects for the indexer
that I think are reasonable and shouldn't cause problems, but I want
to call them up in case something crops up.

This anticipates, somewhat moving the blocksync/statesync/consensus
initialization orchestration into it's own "meta reactor," which I
think we should do, but I think it's not right to do in the context of
this PR.

### CODE REVIEW

File: rpc/client/rpc_test.go [hunk 1]
```diff
@@ -576,10 +577,10 @@ func TestClientMethodCalls(t *testing.T) {
 func getMempool(t *testing.T, srv service.Service) mempool.Mempool {
 	t.Helper()
 	n, ok := srv.(interface {
-		Mempool() mempool.Mempool
+		RPCEnvironment() *rpccore.Environment
```
Comment (creachadair) [inline rpc/client/rpc_test.go:580]: Does this imply we could get rid of the `Mempool()` method too?

**Review by creachadair — APPROVED** [review]

File: rpc/client/rpc_test.go (hunk 1 above)
Comment (tychoish) [reply rpc/client/rpc_test.go:580]: I think I did?

Comment (tychoish) [reply rpc/client/rpc_test.go:580]: yup, it's gone!

Comment (creachadair) [reply rpc/client/rpc_test.go:580]: Oh, oops, I missed it. Yay! 🎉

**Pull request merged**